internal/server/instance/drivers/driver_lxc.go

```diff
@@ -2393,6 +2393,11 @@ func (d *lxc) detachInterfaceRename(netns string, ifName string, hostName string
 
 // Start starts the instance.
 func (d *lxc) Start(stateful bool) error {
+	// require container_migration_stateful is set to true when stateful is true
+	if stateful && util.IsFalse(d.localConfig["migration.stateful"]){
+		return fmt.Errorf("This instance requires migration.stateful to be true. Please set migration.stateful=true on the instance")
```

Comment:
I'd probably rephrase it more like: 

Stateful start requires that the instance migration.stateful set to true